src/Npgsql/BackendMessages/ErrorOrNoticeMessage.cs

```diff
@@ -58,6 +58,10 @@ internal static ErrorOrNoticeMessage Load(NpgsqlReadBuffer buf)
                     break;
                 case ErrorFieldTypeCode.Detail:
                     detail = buf.ReadNullTerminatedStringRelaxed();
+                    if(suppressDetailInPostgressError && string.IsNullOrEmpty(detail) == false)
```

Comment:
It would be better to use the negation operator here rather than comparison, but it's a bit.